R/evaluate-promise.R

```diff
@@ -39,9 +39,12 @@ evaluate_promise <- function(code, print = FALSE) {
   temp <- file()
   on.exit(close(temp))
 
-  result <- with_sink(temp,
-    withCallingHandlers(
-      withVisible(code), warning = wHandler, message = mHandler
+  with_reporter(
```

Comment:
I don't think so, but I'm not in love with this approach - it adds a lot of extra complexity. Let me think on it a bit more.